nvmeadm/src/nvmf_subsystem.rs

```diff
@@ -172,6 +172,24 @@ impl Subsystem {
             Some(subsys) => Ok(subsys),
         }
     }
+
+    /// Gets all Nvme subsystems registered for a given volume.
+    pub fn get_from_nqn(nqn: &str) -> Result<Vec<Subsystem>, NvmeError> {
+        let nvme_subsystems = NvmeSubsystems::new()?;
+        let mut nvme_paths = vec![];
+        for path in nvme_subsystems.flatten() {
+            if path.nqn == nqn && (path.state == "live" || path.state == "connecting") {
```

Comment:
Should we split this into 2 fns, one that gets matching non subsystems, and another than filters it down to only live or connecting paths?

Reply:
Maybe the filtering for live or connecting can be done on the control-plane repo

Reply:
Incorporated